aiohttp/web_urldispatcher.py

```diff
@@ -91,53 +84,113 @@ def _append_query(url, query):
             return url
 
 
-class PlainRoute(Route):
+class ViewableRoute(Route):
+
+    def __init__(self, name, *, expect_handler=None):
+        super().__init__(name, expect_handler=expect_handler)
+        self._views = []
+
+    def match_view(self, request, view_name, match_dict):
+        method = request.method
+        allowed_methods = set()
+
+        for view in self._views:
+            if (view.name == view_name and
```

Comment:
i do not think this is a problem in general. later can add some warning or reordering for view, whatever.